synapse/http/servlet.py

```diff
@@ -295,6 +295,30 @@ def parse_strings_from_args(
         return default
 
 
+@overload
+def parse_string_from_args(
+    args: Dict[bytes, List[bytes]],
+    name: str,
+    default: Optional[str] = None,
+    required: Literal[True] = True,
+    allowed_values: Optional[Iterable[str]] = None,
+    encoding: str = "ascii",
+) -> str:
+    ...
+
+
+@overload
+def parse_string_from_args(
+    args: Dict[bytes, List[bytes]],
+    name: str,
+    default: Optional[str] = None,
+    required: bool = False,
+    allowed_values: Optional[Iterable[str]] = None,
+    encoding: str = "ascii",
+) -> Optional[str]:
```

Comment:
Is this meant to be in here?

Reply:
These changes were necessary since previously the only caller of `parse_string_from_args` that was covered by mypy was `parse_string`, but with the additional type hints being added to the `synapse/federation/transport/server.py` file it seems to have shook out that we need an `overload` for this.